src/component/handlers/composite-android/DraftEditorCompositionHandlerAndroid.js

```diff
@@ -46,6 +48,52 @@ const createMutationObserverIfUndefined = () => {
   }
 };
 
+const startCompositionTimeout = (editor) => {
+  cancelCompositionTimeout();
+
+  if (!getEditorState(editor).isInCompositionMode()) {
+    //editor.update(EditorState.set(getEditorState(editor), { inCompositionMode: true }));
+    editor.silentlyUpdate(EditorState.set(getEditorState(editor), { inCompositionMode: true }));
+  }
+
+  compositionTimeoutId = setTimeout(() => {
+    // If we are at a safe point to exit composition mode, do so to let renders through.
+    const editorState = getEditorState(editor);
+    if (editorState.isInCompositionMode() && isSafeToExitCompositionMode(editor, compositionState)) {
+      DraftEditorCompositionHandlerAndroid.commit(editor, compositionState);
+    }
+  }, compositionTimeoutDelay);
+};
+
+const cancelCompositionTimeout = () => {
+  if (compositionTimeoutId) {
+    clearTimeout(compositionTimeoutId);
+    compositionTimeoutId = null;
+  }
+};
+
+/**
+ * Checks to see if the uncommitted composition state can safely be updated.
+ * If a composition range is updated, the caret _should_ move to the end of the range.
+ * This means we can allow updates through when the caret wouldn't be moved.
+ *
+ * @param {DraftEditor} current editor
+ * @param {EditorState} compositionState
+ */
+const isSafeToExitCompositionMode = (editor, compositionState: EditorState) => {
+  const selection = deriveSelectionFromDOM(editor);
+  const block = compositionState
+        .getCurrentContent()
+        .getBlockForKey(selection.getEndKey());
+
+  const offset = selection.getEndOffset();
+  const char = block.getText()[offset];
+
+  // Check to see if the following character is a non word character, if
+  // it should be safe to allow the composition to change.
+  return (!char || char.match(/\W/));
```

Comment:
It's probably worth a comment that end offsets are exclusive to explain why checking the character at the end offset grabs the character after the selection instead of the last character in the selection.

Reply:
Added comments above this line when setting the offset.